### ISSUE TITLE
Question on overfitting

### ISSUE THREAD

**Issue by JunMa11** [MIC-DKFZ/nnUNet#8]
Dear Fabian,

Thanks for the great work.

I train two 3D fullres U-Nets based on LiTS and my local data (similar to LiTS but only 75 training data). 
> local dataset
![local dataset](http://i1.bvimg.com/595387/22c315d1bf92761e.jpg)
> LiTS
![LiTS](http://i1.bvimg.com/595387/ef1f756221d12a18.jpg)

However, the trained U-Nets have overfitting problem.
Q1. Is there any way to ease the overfitting by adjusting parameters? 

Q2. `nnU-Net` default does not use [dropout](https://github.com/MIC-DKFZ/nnUNet/blob/f70c6ff50b28b2905c5ca563855aaef0a26b93c8/nnunet/training/network_training/nnUNetTrainer.py#L220), right?
If yes, have you tried the nnU-Net with dropout?

Best,
Jun

**Comment by FabianIsensee** [comment]
Dear Jun,
some overfitting will not hurt too much. I have not experimented with dropout, you can add it by modifying 

        dropout_op_kwargs = {'p': 0, 'inplace': True}

in line 220 of nnUNetTrainer.py. This will add dropout to ALL conv layers. If you wish to add it to specific layers only, it will get more complicated.

The easiest way to avoid overfitting is using more training data (duh). There is only so much you can do with data augmentation or other regularization.

I think you can try modifying data augmentation as well to reduce overfitting. Not sure if that will improve your results though.

Best,
Fabian

**Comment by JunMa11** [comment]
Hi Fabian,
Thanks for your quick reply.
I will try it and share the results here.

**Comment by FabianIsensee** [comment]
I have not yet had success with dropout. If you do, definitely let me know :-)

**Comment by JunMa11** [comment]
Hi @FabianIsensee 

Yes, you are right. 
I try `dropout = 0.5` with 2 folds , It takes more time to train (+2 days) for each fold.
Adding dropout can release overfitting a little, but the performance drops obviously.

`fold_0`

|dropout=0|dropout=0.5|
|---|---|
|![without dropout](http://i2.bvimg.com/595387/4b8e09f9a3263115.png)| ![Markdown](http://i2.bvimg.com/595387/da8b0b137c9d35b1.png)|


`fold_2`

|dropout=0|dropout=0.5|
|---|---|
|![Markdown](http://i2.bvimg.com/595387/160fddd4f82e2cc8.png)| ![Markdown](http://i2.bvimg.com/595387/d586de43798abb3a.png)|

**Comment by FabianIsensee** [comment]
Thanks for letting me know and sorry the this did not help :-/

**Comment by JunMa11** [comment]
Hi @FabianIsensee ,
You do not need to say sorry, I really enjoy nnUNet and your code comments:-)
I use nnUNet on my local dataset, the performance exceeds all my previous experiments.
I really appreciate that you make this great work public available.

**Comment by FabianIsensee** [comment]
Thank you so much, it warms my heart to hear that you enjoy it and that the repo runs so well on your data!
